containerize/Program.cs

```diff
@@ -1,73 +1,124 @@
+            return new string[] { };
+        }
+        return result.Tokens.Select(v => v.Value).ToArray<string>();
+    });
+
+RootCommand root = new RootCommand("Containerize an application without Docker.")
+{
+    publishDirectoryArg,
+    baseRegistryOpt,
+    baseImageNameOpt,
+    baseImageTagOpt,
+    outputRegistryOpt,
+    imageNameOpt,
+    imageTagsOpt,
+    workingDirectoryOpt,
+    entrypointOpt,
+    entrypointArgsOpt,
+    labelsOpt
 };
-rootCommand.SetHandler(async (folder, containerWorkingDir, uri, baseImageName, baseTag, entrypoint, imageName, imageTag) =>
+
+root.SetHandler(async (context) =>
 {
-    await Containerize(folder, containerWorkingDir, uri, baseImageName, baseTag, entrypoint, imageName, imageTag);
-},
-    fileOption,
-    workingDir,
-    registryUri,
-    baseImageName,
-    baseImageTag, 
-    entrypoint,
-    imageName,
-    imageTag
-    );
-
-return await rootCommand.InvokeAsync(args);
-
-async Task Containerize(DirectoryInfo folder, string workingDir, string registryName, string baseName, string baseTag, string[] entrypoint, string imageName, string imageTag)
+    DirectoryInfo _publishDir = context.ParseResult.GetValueForArgument(publishDirectoryArg);
+    string _baseReg = context.ParseResult.GetValueForOption(baseRegistryOpt) ?? "";
+    string _baseName = context.ParseResult.GetValueForOption(baseImageNameOpt) ?? "";
+    string _baseTag = context.ParseResult.GetValueForOption(baseImageTagOpt) ?? "";
+    string _outputReg = context.ParseResult.GetValueForOption(outputRegistryOpt) ?? "";
+    string _name = context.ParseResult.GetValueForOption(imageNameOpt) ?? "";
+    string[] _tags = context.ParseResult.GetValueForOption(imageTagsOpt) ?? Array.Empty<string>();
+    string _workingDir = context.ParseResult.GetValueForOption(workingDirectoryOpt) ?? "";
+    string[] _entrypoint = context.ParseResult.GetValueForOption(entrypointOpt) ?? Array.Empty<string>();
+    string[] _entrypointArgs = context.ParseResult.GetValueForOption(entrypointArgsOpt) ?? Array.Empty<string>();
+    string[] _labels = context.ParseResult.GetValueForOption(labelsOpt) ?? Array.Empty<string>();
+
+    await Containerize(_publishDir, _workingDir, _baseReg, _baseName, _baseTag, _entrypoint, _entrypointArgs, _name, _tags, _outputReg, _labels);
+});
+
+return await root.InvokeAsync(args);
+
+async Task Containerize(DirectoryInfo folder, string workingDir, string registryName, string baseName, string baseTag, string[] entrypoint, string[] entrypointArgs, string imageName, string[] imageTags, string outputRegistry, string[] labels)
 {
-    Registry registry = new Registry(new Uri($"http://{registryName}"));
+    Registry registry = new Registry(new Uri(registryName));
 
     Console.WriteLine($"Reading from {registry.BaseUri}");
 
-    Image x = await registry.GetImageManifest(baseName, baseTag);
-    x.WorkingDirectory = workingDir;
+    Image img = await registry.GetImageManifest(baseName, baseTag);
```

Comment:
For a lot of this, I was kinda hoping we could extract a library method from the existing task code, so that `containerize` is just a command-line→library-method adapter, and the task is just an MSBuild-stuff→library-method adapter. Think that's doable?

Reply:
Doing so leaves a hole in our logging. My helper method today just Console.WriteLines. We could have it accept a TaskLoggingHelper optionally, and potentially log to console if it's null (being called from Containerize)